### PR TITLE
Improve SearchPage error handling

### PR DESCRIPTION
**Pull request by edalex-ian**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]

##### Description of change

A number of possible errors were not being caught. This would result in
spurious console log entries, and lack of feedback to the user that an
issue had occurred.

Most of these are now addressed, by simply shutting down oEQ and then
chasing things popping up in the console log. Or in a few cases, just by
manual code analysis and looking at the user result.

And also snuck in a minor IntelliJ project file update.

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: Source/Plugins/Core/com.equella.core/js/tsrc/search/SearchPage.tsx [hunk 1]
```diff
@@ -158,24 +162,29 @@ const SearchPage = ({ updateTemplate }: TemplateUpdateProps) => {
...
-      else
-        setSearchPageOptions({
-          ...searchPageOptions,
-          sortOrder:
-            searchPageOptions.sortOrder ?? searchSettings.defaultSearchSort,
-        });
-    });
+    ])
+      .then((results) => {
+        const [searchSettings, queryStringSearchOptions] = results;
```
Comment (ChristianMurphy) [inline Source/Plugins/Core/com.equella.core/js/tsrc/search/SearchPage.tsx:167]:
```suggestion
      .then(([searchSettings, queryStringSearchOptions]) => {
```

**Review by ChristianMurphy — APPROVED** [review]
Looks good, happy to see more error handling, thanks @edalex-ian!

---

related to this, though likely in scope for this PR.
To ensure consistent catching of promises in the future eslint plugin promise https://github.com/xjamundx/eslint-plugin-promise#rules may be able to help

**Review by PenghaiZhang — APPROVED** [review]
Thanks Ian! Looks good to me!

File: Source/Plugins/Core/com.equella.core/js/tsrc/search/SearchPage.tsx (hunk 1 above)
Comment (edalex-ian) [reply Source/Plugins/Core/com.equella.core/js/tsrc/search/SearchPage.tsx:167]: Oh that's an interesting one. That's just part of the prettier reformat. I'll have a play though and see if things are still happy or whether @PenghaiZhang did this for a reason - I've just got a faint memory of something like this coming up before. :thinking:

Comment (PenghaiZhang) [reply Source/Plugins/Core/com.equella.core/js/tsrc/search/SearchPage.tsx:167]: The array destructuring? I think Charlie added this when he worked on converting query string to SearchOption.

**Comment by edalex-ian** [comment]
Will just wait for a green build and the merge.

File: Source/Plugins/Core/com.equella.core/js/tsrc/search/SearchPage.tsx (hunk 1 above)
Comment (edalex-ian) [reply Source/Plugins/Core/com.equella.core/js/tsrc/search/SearchPage.tsx]: Yeah, once I was tweaking it I realised it was newer. And seemed all good. :+1:

**Pull request merged**